src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -763,8 +780,13 @@ fn do_mul(lhs: &Decimal, rhs: &Decimal) -> Res<Decimal> {
     let mut start_to = int_word_to + frac_word_to;
     let r_start = cmp::max(r_int_word_cnt + r_frac_word_cnt, 0) as usize;
     let left_stop = cmp::max(l_int_word_cnt + l_frac_word_cnt, 0) as usize;
+    let r_start = cmp::min(r_start, WORD_BUF_LEN as usize);
+    let left_stop = cmp::min(left_stop, WORD_BUF_LEN as usize);
     for l_idx in (0..left_stop).rev() {
-        assert!(start_to >= r_start);
+        if start_to < r_start {
+            break;
```

Comment:
Why it is not assert any more?

Reply:
Since start_to may be smaller than r_start, and the loop should be break when it happens.